### DISCUSSION TITLE
help request: How to setup http_upgrade  on apisix dashboard

### DISCUSSION THREAD

**Discussion by xerox-xeon** [apache/apisix#8177] [Show and tell]
### Description
like this: 
if ($http_upgrade !~* ^WebSocket$) {
                return 404;
            }

### Environment

- APISIX version (run `apisix version`):
- Operating system (run `uname -a`):
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

**Comment by tzssangglass** [comment]
we can use `lua-resty-expr`: https://github.com/api7/lua-resty-expr, it may be like:

```
"vars": [["http_user_upgrade", "~=", "WebSocket"]]
```

in APISIX Dashboard, `lua-resty-expr` show as `高级匹配条件`
